packages/dynamic-import-vars/src/dynamic-import-to-glob.js

```diff
@@ -98,9 +114,29 @@ export function dynamicImportToGlob(node, sourceString) {
   }
 
   if (!glob.startsWith('./') && !glob.startsWith('../')) {
-    throw new VariableDynamicImportError(
-      `invalid import "${sourceString}". Variable bare imports are not supported, imports must start with ./ in the static part of the import. ${example}`
-    );
+    const match = glob.match(npmPackageNameAndGlobRegex);
+
+    if (match) {
+      const [, packageName, , packageGlob] = match;
+      const resolvedPackageJson = await r.resolve(`${packageName}/package.json`, id);
+
+      if (!resolvedPackageJson) {
+        throw new VariableDynamicImportError(
+          `invalid import "${sourceString}". Could not resolve module "${packageName}".`
+        );
+      }
+
+      glob = `${path.relative(
+        path.dirname(id),
+        path.dirname(resolvedPackageJson.id)
+      )}/${packageGlob}`;
+      bareImport = packageName;
+      resolvedBareImport = path.dirname(resolvedPackageJson.id);
+    } else {
+      throw new VariableDynamicImportError(
+        `invalid import "${sourceString}". Non-relative variable imports that do not match an NPM package name are not supported, imports must start with either an npm package name or ./ in the static part of the import. ${example}`
```

Comment:
I'm not keen on `an NPM package name` bit here. I think a more accurate, registry agnostic string such as `an installed Node package` would be better